### ISSUE TITLE
WebApp:az webapp update raised Exception

### ISSUE THREAD

**Issue by drjayvee** [Azure/azure-cli#16208]
## Describe the bug

**Command Name**
`az webapp update --name sw33t-d3m0 -gtehst -ntehstas`

**Errors:**
```
'NoneType' object has no attribute 'kind'
Traceback (most recent call last):
python3.6/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 717, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 710, in _run_job
    six.reraise(*sys.exc_info())
...
cli/core/commands/__init__.py, ln 325, in __call__
    return self.handler(*args, **kwargs)
cli/core/commands/arm.py, ln 527, in handler
    instance = custom_function(instance=instance, **custom_func_args)
cli/command_modules/appservice/custom.py, ln 629, in update_webapp
    if 'function' in instance.kind:
AttributeError: 'NoneType' object has no attribute 'kind'
```

## To Reproduce:
Follow the [tutorial](https://docs.microsoft.com/en-us/azure/app-service/tutorial-php-mysql-app?pivots=platform-linux#create-a-web-app) to create a webapp with awful name (`sw33t-d3m0`)

## Environment Summary
```
Linux-4.15.0-1100-azure-x86_64-with-debian-10.2 (Cloud Shell)
Python 3.6.10
Installer: DEB

azure-cli 2.16.0

Extensions:
ai-examples 0.2.5

```
## Additional Context
I first ran
```
$ az webapp update --name sw33t-d3m0 -ntehstas
(--resource-group | --ids) are required
```
I had seen in a different command that `--resource-group` has a shorthand `-g`, which I used. Using the long option results in the same error.

I realize now that I provided both `--name` and `-n`, which is a bit weird.

**Comment by yungezz** [comment]
routing to appropriate team

**Comment by yungezz** [comment]
#14287

**Comment by ThejaChoudary** [comment]
@drjayvee When using CLI make sure the subscription you are using is the one that has the RG you are using in the command.You can az account show to verify this. This error indicates that App or RG values are either incorrect or are not on the subscription that is being used.

**Comment by drjayvee** [comment]
Hi there. I've actually stopped working on Azure, so I won't delve into this issue any farther.

I'll leave it up to you all whether the CLI should provide a clear error message or not.

**Comment by panchagnula** [comment]
closing since this should be fixed by #14287
